### PR TITLE
Refo 106: Admin Job dashboard and employer dashboard

### PR DESCRIPTION
**Pull request by makersmelx**
This the entire part for the job management. Including all features that are listed in figma. 
- Sorry i do not have enough time to make the code organized.
- Just test all the features and workflow we expected to have. And when you find a bug just reach out to me
- With the `edit` `approve` `reject` button, the page height, flex part may be broken again in some conditions. No time to fix it
- If I, by mistake, revert your work, it's because of the Github merge issue. Let me know and I will fix this back.
- [463f4dd](https://github.com/RefugeeOne699/refugee-one/pull/57/commits/463f4dd23b7f3ca18409aea46be5afd3d34eadb2) @kendebacker  this is how the delete job works. When you delete a user, refer to this as how to delete job ownership, and job save relations
- [698048f](https://github.com/RefugeeOne699/refugee-one/pull/57/commits/698048f6f34412c13f8fd42b1212c78e19f1c685) @kendebacker this is an example of how the buttons should behave when a request is running

### CODE REVIEW

File: src/components/job/jobActions/AdminActions.jsx [hunk 1]
```diff
@@ -1,53 +1,71 @@
...
+          return;
+        }
+        toast.success(`Job has been ${params[0] ? "approved" : "rejected"}!`);
+        await dashboard.jobsRefresh();
+        await dashboard.countRefresh();
+        navigate("..", { replace: true });
+      },
+      onError: (error, params) => {
+        toast.error(
+          `Failed to ${params[0] ? "approve" : "rejecte"} job: ${error.message}`
```
Comment (kendebacker) [inline src/components/job/jobActions/AdminActions.jsx:40]: "rejected"

Comment (kendebacker) [reply src/components/job/jobActions/AdminActions.jsx:40]: @makersmelx

File: src/components/job/JobDashboard.jsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import { toast } from "react-hot-toast";
+import { NavLink, useParams } from "react-router-dom";
+
+import JobRoot from "@/components/job/JobRoot";
+import Spin from "@/components/Spin";
+import { JOB_STATUS } from "@/constants";
+import { useJob } from "@/models";
+
+const menu = [
+  { url: "pending", name: "All Pending Jobs" },
```
Comment (kendebacker) [inline src/components/job/JobDashboard.jsx:13]: use constants for url?

**Review by kendebacker — APPROVED** [review]

File: src/components/job/JobDashboard.jsx (hunk 2 above)
Comment (makersmelx) [reply src/components/job/JobDashboard.jsx:13]: no need. `constants` are for which are shared in multiple files

File: src/components/job/UpsertJob.jsx [hunk 3]
```diff
@@ -1,30 +1,78 @@
...
+      if (jobData) {
+        Object.entries(data).forEach(([key, value]) => {
+          jobData[key] = value;
+        });
+        return job.updateJob(jobId, jobData);
+      }
+      // create
       return job.createJob({
         ...data,
         adminMessage: "",
```
Comment (KiLiG-fcx) [inline src/components/job/UpsertJob.jsx:78]: Location field (string concat from address field) is not updated, so job view still showing the old data.

**Review by nehakumari2307 — APPROVED** [review]

**Pull request merged**